File: programmers/월간 코드 챌린지 시즌2/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0
    for num in range(left, right + 1, 1):
        num_cnt = len([i for i in range(1, num + 1) if num % i == 0])
        if num_cnt % 2:
            answer -= num
        else:
            answer += num
    return answer

File: programmers/월간 코드 챌린지 시즌2/약수의 개수와 덧셈.kt
// Solution 1
class Solution {
    fun solution(left: Int, right: Int): Int {
        var answer: Int = 0
        for (i in left..right) {
            var cnt = (1..i).toList().toIntArray().filter {
                i % it == 0
            }.count()
            if (cnt % 2 == 0) { // 짝수인 경우
                answer += i
            } else {
                answer -= i
            }
        }
        return answer
    }
}

// Solution 2 (refactoring)
class Solution {
    fun solution(left: Int, right: Int): Int {
        return (left..right).map {
            i -> if ((1..i).filter {i % it == 0}.size % 2 == 0) i else -i
        }.sum()
    }
}